src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -309,37 +309,33 @@ private static ulong GetKey(ReadOnlySpan<byte> propertyName)
 
         public static Type GetElementType(Type propertyType)
         {
-            Type elementType = null;
             if (typeof(IEnumerable).IsAssignableFrom(propertyType))
             {
-                elementType = propertyType.GetElementType();
-                if (elementType == null)
+                Type elementType = propertyType.GetElementType();
+                if (elementType != null)
                 {
-                    Type[] args = propertyType.GetGenericArguments();
+                    return elementType;
+                }
+
+                Type[] args = propertyType.GetGenericArguments();
```

Comment:
Should this be inside the `if (propertyType.IsGenericType)`?

Reply:
Yes thanks.